### ISSUE TITLE
[Question] Deployment strategy when dealing with multiple big databases

### ISSUE THREAD

**Issue by codref** [babelfish-for-postgresql/babelfish_extensions#2226]
While migrating to Babelfish (which so far is great!!) I ended up with a dilemma, what's the best deployment strategy when SQL Server was hosting several (>5) big databases (from 500GB to 1TB each).  

The plan would be to progressively move to Postgres in the future, therefore the `single-db` seems the best choice, but this means *running multiple Babelfish instances on the same server, each on a different ports* (e.g 1433, 1434, 1435, etc).  

Are there issues in running multiple Postgres instances on the same OS?  
Could this make backups easier?  
The lack of easy users management (so far mitigated only with sysadmin users) also seems to apply better to the `single-db` mode, is this true?  

Thanks!

**Comment by sksql** [comment]
To achieve the "SQL Server experience" with Babelfish, I strongly recommend multi-db mode. Having said that, if you're objective/plan is to migrate off Babelfish to PostgreSQL pretty soon then single-db mode makes sense but IMO that's the only scenario.

Do not use pg_dump for backups as it's not supported for Babelfish. The world wide internet may have advise; personally I recommend bbf_dump/bbf_restore - watch this space.

**Comment by codref** [comment]
Thanks! I was not aware of bbf_dump/bbf_restore commands, there is no documentation (or is hard to find), so I do believe your reply can be useful to others as well!  
I will leave this open and close as soon as I verify, to provide a complete answer.

**Comment by sksql** [comment]
The utilities are coming, so there's no documentation yet. But watch this space.

**Comment by staticlibs** [comment]
We use a number of small (usually 5-10 GB) DBs with MSSQL and need to continue supporting MSSQL along with Babelfish, so using only multi-db mode. After experimenting quite a bit with Babelfish-flavoured pg_dump/restore (added some notes about it [here](https://github.com/wiltondb/wiltondb/wiki/Backup-and-restore-overview-in-Babelfish)) they seem very close to be fully working. Hit a number of minor bumps (like lack of DB rename), going to re-check it with the latest Babelfish (we stick with 3.3) and file issues here about some of them.

**Comment by suprio-amzn** [comment]
> _The utilities are coming, so there's no documentation yet. But watch this space._

The documentation is finally out: https://github.com/babelfish-for-postgresql/babelfish-for-postgresql/wiki/Babelfish-dump-and-restore

**Comment by codref** [comment]
This is great, thanks for sharing! The `--roles-only` dump seems much more convenient than the manual way I was following!  
I tested this on DBs with size > 400GB and I have to say that the dump is fast and efficient, the restore in my case was very slow - as I followed the WiltonDB how-to where the pg_restore was with `-j 1` flag set. This seems over cautious and not needed (according to new official documentation) so one problem less!!  

In the end I opted for multiple instances running on multiple docker containers on the same server.  
This way I can workaround the limitations on users permissions and grant full access at DB level (to multiple users).  
I'm not super sure about resources use during peak concurrency - and I struggle to find any good documentation on the subject.  
So far this project is amazing, and I'd love to support you guys!
